#!/usr/bin/env python

PACKAGE = "Final_Project"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
"""
gen.add("int_param",    int_t,    0, "An Integer parameter",    50,  0, 100)
gen.add("double_param",    double_t,    0, "A double parameter",    .5, 0,   1)
gen.add("str_param",    str_t,    0, "A string parameter",    "Hello World")
gen.add("bool_param",    bool_t,    0, "A Boolean parameter",    True)
"""
"""
gen.add("x_des",	double_t,	0,	"X desired",	0, -1, 1)
gen.add("y_des",	double_t,	0,	"Y desired",	0, -1, 1)
gen.add("z_des",	double_t,	0,	"Z desired",	0, -1, 1)
"""

gen.add("P_XY_1", double_t, 0, "Mode 1: Proportional constant of the PID controller", 0.04, -1, 1)
gen.add("I_XY_1", double_t, 0, "Mode 1: Integral constant of the PID controller", 0.02, -1, 1)
gen.add("D_XY_1", double_t, 0, "Mode 1: Derivative constant of the PID controller", 0.2, -1, 1)
gen.add("P_Z_1", double_t, 0, "Mode 1: Proportional constant for Z axis", 0.3, -1, 1)
#gen.add("I_Z_1", double_t, 0, "Mode 1: Integral constant for Z axis", 0, -1, 1)
#gen.add("D_Z_1", double_t, 0, "Mode 1: Derivative constant for Z axis", 0, -1, 1)
gen.add("P_Yaw_1", double_t, 0, "Mode 1: Proportional constant for rotation in Yaw", 0.08, -1, 1)
#gen.add("I_Yaw_1", double_t, 0, "Mode 1: Integral constant for rotation in Yaw", 0, -1, 1)
gen.add("D_Yaw_1", double_t, 0, "Mode 1: Derivative constant for rotation in Yaw", 0.1, -1, 1)
gen.add("P_X_2", double_t, 0, "Mode 2: Proportional constant of the PID controller", 0.01, -1, 1)
gen.add("D_X_2", double_t, 0, "Mode 2: Derivative constant of the PID controller", 0.03, -1, 1)
gen.add("P_Y_2", double_t, 0, "Mode 2: Proportional constant of the PID controller", 0.025, -1, 1)
gen.add("I_Y_2", double_t, 0, "Mode 2: Integral constant of the PID controller", 0, -1, 1)
gen.add("D_Y_2", double_t, 0, "Mode 2: Derivative constant of the PID controller", 0.1, -1, 1)
gen.add("P_Z_2", double_t, 0, "Mode 2: Proportional constant for Z axis", 0.1, -1, 1)
#gen.add("I_Z_2", double_t, 0, "Mode 2: Integral constant for Z axis", 0, -1, 1)
#gen.add("D_Z_2", double_t, 0, "Mode 2: Derivative constant for Z axis", 0, -1, 1)
gen.add("P_Yaw_2", double_t, 0, "Mode 2: Proportional constant for rotation in Yaw", 0.06, -1, 1)
gen.add("I_Yaw_2", double_t, 0, "Mode 2: Integral constant for rotation in Yaw", 0.01, -1, 1)
gen.add("D_Yaw_2", double_t, 0, "Mode 2: Derivative constant for rotation in Yaw", 0, -1, 1)

"""
size_enum = gen.enum([ gen.const("Small",    int_t, 0, "A small constant"),
                                         gen.const("Medium",    int_t, 1, "A medium constant"),
                                         gen.const("Large",    int_t, 2, "A large constant"),
                                         gen.const("ExtraLarge",    int_t, 3, "An extra large constant")],
                                        "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)
"""

exit(gen.generate(PACKAGE, "Final_Project", "controller"))
